### ISSUE TITLE
calc_chain should raise an exception if user sets a variable not in the data pool

### ISSUE THREAD

**Issue by billbrod** [WinawerLab/sco#14]
From typos and sloppiness, it's possible to accidentally set a variable not in the data pool (e.g., `pRF_blob_std` instead of `pRF_blob_stds`). We should probably raise an exception, because it generally means the user is not running the model they think they are

**Comment by noahbenson** [comment]
So I think there are good reasons not to do this, or at least good reasons not to do it all the time. If you are running a calculation step-by-step, for example, your datapool will contain things from other steps at any given time, so you'd have to pull out only the relevant items. I do this pretty frequently when debugging. Maybe we should instead add a way to validate a parameter set against a calculation so that you can check for errors in the parameter set before running it?

**Comment by billbrod** [comment]
Oh yeah, that makes sense. 

Someway to validate the parameter set would work. Or maybe have a toggle so that this check happens by default and you only don't do this check if the user specifically wants to skip it? (while they're debugging or the like). I guess the question is, what's the user-case? Will people normally be doing the whole chain all at once, only doing it step by step to debug? Or will they frequently do it step by step as well?

**Comment by noahbenson** [comment]
Oh! Or we could do the thing that's too obvious to imagine: raise a warning

**Comment by billbrod** [comment]
That's true. Let's do that for now and then think about it more later to see if we want to do something more stringent? I can add that warning.

**Comment by billbrod** [comment]
Turns out, adding that warning is not super easy: because of the modularity of the calc chain, it doesn't know when you're entering a chain for the first time or when you're entering it from another chain. So we'll need to do some thinking about how best to do this. Noah's going to restructure some of the code soon, so we won't do anything until then.
